tests/integration/eventing/userDefinedMessage.ts

```diff
@@ -0,0 +1,162 @@
+      aliceCall.on('accept', () => {
+        res();
+      });
+    });
+    const bobCallAcceptPromise = new Promise((res) => {
+      bobCall.on('accept', () => {
+        res();
+      });
+    });
+
+    bobCall.accept();
+
+    await aliceCallAcceptPromise;
+    await bobCallAcceptPromise;
+
+    const performTeardown = () => {
+      aliceDevice.destroy();
+      bobDevice.destroy();
+    };
+
+    return {
+      performTeardown,
+      alice: {
+        device: aliceDevice,
+        call: aliceCall,
+      },
+      bob: {
+        device: bobDevice,
+        call: bobCall,
+      },
+    };
+  };
+
+  describe('outgoing call leg', function() {
+    it(
+      'should successfully send a message to the customer server',
+      async function() {
+        const { performTeardown, alice: { call } } = await createCallTest();
+
+        const CallSid = call.parameters.CallSid;
+        console.log('call sid', CallSid);
+        const subscription = await axios.post(
+          `${RELAY_SERVER_URL}/create-subscription`,
+          { CallSid },
+        );
+        console.log(`subscription made for call ${CallSid}`, subscription.data);
+
+        const eventSid = call.sendMessage({
+          content: { hello: 'world' },
+          messageType: 'user-defined-message',
+        });
+        await new Promise((res, rej) => {
+          console.log('waiting for twilio to receive the message')
+          call.once('messageSent', (msg) => {
+            if (msg.voiceEventSid === eventSid) {
+              return res();
+            }
+            rej();
+          });
+        });
+        console.log('sent message', eventSid);
+
+        // NOTE(mhuynh): Using a quick solution here such that we just wait for
+        // the message from Twilio to reach the test server. Otherwise, there is
+        // a high likelihood that the message will be missed.
+        await new Promise((res) => {
+          setTimeout(() => {
+            res();
+          }, 3000);
+        });
+
+        const receivedMessagesResponse = await axios.post(
+          `${RELAY_SERVER_URL}/get-received-messages`,
+          { CallSid },
+        );
+        const receivedMessages = receivedMessagesResponse.data;
+        console.log('received messages', receivedMessages);
+
+        assert.strictEqual(typeof receivedMessages, 'object');
+        assert(
+          Array.isArray(receivedMessages),
+          'received messages is not of type array',
+        );
+        assert.strictEqual(receivedMessages.length, 1);
+
+        const [receivedMessage] = receivedMessages
+        assert.strictEqual(receivedMessage.ContentType, 'application/json');
+        assert.strictEqual(receivedMessage.Content, JSON.stringify({
+          hello: 'world',
+        }));
+        assert.strictEqual(receivedMessage.SequenceNumber, '1');
+        assert.strictEqual(receivedMessage.CallSid, CallSid);
+        assert.strictEqual(receivedMessage.Sid, eventSid);
```

Comment:
Use `assert.deepStrictEqual` so you don't have to do multiple assertions.

Reply:
There's some members that I didn't want to pull in, so I just opted to do each member individually. For instance, there's a timestamp value that is part of the larger object, and I can't do a `deepStrictEqual` match on that.